test/Base_Tests/src/Data/Array_Proxy_Spec.enso

```diff
@@ -43,7 +43,7 @@ add_specs suite_builder =
 
         group_builder.specify "should check validity at construction" <|
             Test.expect_panic_with (Array_Proxy.new 0 0) Type_Error
-            Test.expect_panic_with (Array_Proxy.new -1 (x->x)) Illegal_Argument
+            (Array_Proxy.new -1 (x->x)).should_fail_with Illegal_Argument
```

Comment:
I have now fixed the `Test.expect_panic_with` to only intercept **actual Panics**. Before it used to happily catch dataflow errors as well and not distinguish between them.

Our tests should distinguish between the two cases.

After my changes this test used to fail with:
![image](https://github.com/enso-org/enso/assets/1436948/ea72cd33-64a0-4c21-ab26-c16e4c2a164c)